core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1042,8 +1042,19 @@ public GHResponse route(GHRequest request) {
             // For example see #734
             checkIfPointsAreInBounds(points);
 
-            // TODO NOW for CH it will be overwritten by chProfile.getWeighting but now we need it for locationIndex -> #729
-            Weighting weighting = createWeighting(hints, encoder);
+            RoutingAlgorithmFactory algoFactory = getAlgorithmFactory(hints);
+            Weighting weighting;
+            CHProfile chProfile = null;
+            if (chFactoryDecorator.isEnabled() && !disableCH) {
+                if (algoFactory instanceof CHRoutingAlgorithmFactory) {
```

Comment:
I removed the feature that LM can be combined with CH as we currently do not use it (and it is not faster for node-based). Or should I include it as it is a bit worth for edge-based?

Reply:
Have you tried it with edge-based ? Currently we are using a straight-line distance heuristic for edge-based CH AStar, not sure if its an improvement to use LM in this case. But if we never used it so far, sure why not remove it. I like this because it reduces some of the factory/decorator complexity.

Reply:
Hmmh. Actually the decorator stuff was only introduced due to this use case (LM+CH), so will probably not remove it here and a separate refactoring is better.

Reply:
Hmmh, yes. Actually the decorator stuff was only introduced due to this use case (LM+CH) if I correctly remember, so will probably not remove it here and a separate refactoring is better.